### PR TITLE
getCustomPragmaVal priority/override fixes

### PR DESCRIPTION
**Pull request by Clyybber**

### CODE REVIEW

File: lib/core/macros.nim [hunk 1]
```diff
@@ -1536,6 +1536,14 @@ proc getPragmaNodeFromTypeSym(sym: NimNode): NimNode =
     if pragmaExpr.kind == nnkPragmaExpr:
       result = pragmaExpr[1]
 
+proc getPragmaNodeFromType(typ: NimNode): NimNode =
+  case typ.kind:
```
Comment (Araq) [inline lib/core/macros.nim]:
```suggestion
  case typ.kind
```

**Comment by ghost** [comment]
Not sure if the test failure is directly related to nimpylib - you can reproduce it with this:
```nim
import std/macros
```
and compiling it with `nim c --experimental:strictFuncs file.nim`

**Comment by ghost** [comment]
Output is:
```c
/home/dian/Things/Nim/lib/core/macros.nim(1588, 69) template/generic instantiation of `&` from here
/home/dian/Things/Nim/lib/system.nim(1697, 8) Error: '&' can have side effects
an object reachable from 'x' is potentially mutated
/home/dian/Things/Nim/lib/system.nim(1711, 23) the mutation is here
/home/dian/Things/Nim/lib/system.nim(1711, 23) is the statement that connected the mutation to the parameter
```

**Comment by Araq** [comment]
We need a `system.nim` fix for `&`, it never has side effects.

**Pull request merged**